### DISCUSSION TITLE
Communicating environment

### DISCUSSION THREAD

**Discussion by kit-calloway** [CMP2804/Assessment2#1] [Ideas]
The game's protagonist is a master thief, recently blinded by an an accident.
Gameplay largely revolves around trying to navigate the environment with limited senses.
Because the protagonist is blind, they must use their other senses to visualise their environment.
The protagonist can make sound by clapping their hands (loud), clicking their fingers (quiet) and by throwing objects (suspicious). The louder the sound, the clearer the environment becomes. However, this presents an increased risk as these sounds can attract guards.
We can explore the usage of point clouds to achieve this effect.
Scanner Sombre and Watch Dogs use this effect:

In particular, throwing objects can raise suspicion as objects may be out-of-place.
![image](https://user-images.githubusercontent.com/101411609/213867032-992cfbad-481c-4046-b142-3d1613ceacde.png)
![image](https://user-images.githubusercontent.com/101411609/213867066-4ac41086-0e30-4daf-9f0c-15dd5dcd065b.png)
![image](https://user-images.githubusercontent.com/101411609/213867098-dcef952a-8bd5-4842-ad8c-62b73029f69c.png)

**Comment by kit-calloway** [comment]
Here's what I have so far:
- point clouds
- clicking & throwing to see environment
- loud sounds attract enemies
- interactive objects could be highlighted separate colours
- environment would be clearer around louder objects

**Comment by Vesper-Works** [comment]
Should the environment be ambiently lit, or should the player be in total darkness?

**Reply by kit-calloway** [reply]
Given that the character is blind, light would have no effect anyway.

**Reply by Vesper-Works** [reply]
I don't mean literal light, but rather a faint vision of the surroundings

**Reply by Vesper-Works** [reply]
I reckon objects touched by the player should be added to the point cloud also

**Comment by HFinch13** [comment]
could we have outside areas? denoted by more ambient noise, such as wind or vehicle engines. these would more likely be connecting areas not actual focal points of the map

**Reply by kit-calloway** [reply]
We could, I suppose.

**Reply by Vesper-Works** [reply]
Since you're designing the level I'd say go for it!

**Comment by kit-calloway** [comment]
I think a small amount of ambient vision should be visible - if you've ever played Barotrauma with the sonar disabled, you can see big objects but it's rather unreliable.

**Reply by Vesper-Works** [reply]
Could you find an image of what that looks like?

**Reply by kit-calloway** [reply]
Try here:
https://youtu.be/AZLQojocR1k?t=540

**Reply by Vesper-Works** [reply]
Hmm I don't really see it very well
